healthcheck/tor_connection.go

```diff
@@ -0,0 +1,85 @@
+package healthcheck
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"syscall"
+
+	"github.com/lightningnetwork/lnd/tor"
+)
+
+// CheckTorServiceStatus checks the Onion Service's info to decide whether the
+// Tor daemon is up and running. We will get an EOF or a broken pipe error if
+// the Tor daemon is stopped/restarted as the previously created socket
+// connection is no longer open. In this case, we will attempt a restart on our
+// tor controller. If the tor daemon comes back, a new socket connection will
+// then be created.
+func CheckTorServiceStatus(tc *tor.Controller, createService func() error) error {
+	err := tc.GetServiceInfo()
```

Comment:
Ah ok question answered re prior comment, maybe change the name to reflect this is itself more of a health check and we don't actually parse the response?

Reply:
The function is now renamed to `CheckOnionService`